test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -3994,6 +3994,9 @@ public async Task DurableEntity_AsyncIO()
                 await client.SignalEntity(this.output, "get", "https://www.microsoft.com");
                 await client.SignalEntity(this.output, "get", "https://bing.com");
 
+                // Sleeping for a short time here appears to ensure that the state is fully updated before the read
+                Thread.Sleep(500);
+
                 var state = await client.WaitForEntityState<IDictionary<string, string>>(this.output, TimeSpan.FromSeconds(10));

```

Comment:
I'm wondering, could the `client.WaitForEntityState(...)` API have been used to account for these delays?